File: 백준/Silver/1406. 에디터/에디터.py
import sys
input = sys.stdin.readline

word = list(input().strip())
word_stack = []
tc_num = int(input())
cursor = len(word)

for tc in range(tc_num):
    order = input().split()

    if order[0] == 'L' and word:
        word_stack.append((word.pop()))

    elif order[0] == 'D' and word_stack:
        word.append(word_stack.pop())

    elif order[0] == 'B' and word:
        word.pop()

    elif order[0] == 'P':
        word.append(order[1])


word.extend(reversed(word_stack))

for char in word:
    print(char, end='')

File: TypeScript/백준/Silver/1406. 에디터/에디터.ts
const readline = require("readline");

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input: string[] = [];

rl.on("line", (line) => {
  // 입력 전처리
  input.push(line);
});

rl.on("close", () => {
  // 메인 로직
  const [targetString, _, ...orderList] = input;

  const left: string[] = targetString.split("");
  const right: string[] = [];

  orderList.forEach((orderCom) => {
    const [order, addString] = orderCom.split(" ");
    switch (order) {
      case "L":
        if (left.length > 0) {
          right.push(left.pop());
        }
        break;
      case "D":
        if (right.length > 0) {
          left.push(right.pop());
        }
        break;
      case "B":
        left.pop();
        break;
      case "P":
        left.push(addString);
        break;
      default:
        throw new Error("Exhaustive Checked, Wrong Order");
    }
  });

  console.log(left.concat(right.reverse()).join(""));
  process.exit();
});
